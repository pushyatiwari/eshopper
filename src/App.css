.App {
  text-align: center;
  display: flex;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}


.container{
      display: flex;
      flex-direction: row;
  }
  @media only screen and (max-width: 20rem) {
      .rendercards{
          flex-direction: column;
          align-content: stretch;
          flex-wrap: wrap;
      }
  }