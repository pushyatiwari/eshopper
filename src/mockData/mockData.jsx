const axiosDataGetItems = {
  data: {
    data: [
      {
        id: 1,
        name: 'apple',
        price: 120,
        count: 20,
        category: 'Fruits & Vegatables',
      },
      {
        id: 2,
        name: 'table cloth',
        price: 200,
        count: 3,
        category: 'Household Items',
      },
    ],
  },

};

const axiosDataGetOrders = {
  data: {
    data: [
      {
        items: [
          {
            id: 1,
            name: 'apple',
            price: 120,
            count: 1,
            category: 'Fruits & Vegatables',
          },
        ],
        id: 1,
        date: 1615122360481,
      },
      {
        items: [
          {
            id: 2,
            name: 'table cloth',
            price: 200,
            count: 1,
            category: 'Household Items',
          },
          {
            id: 9,
            name: 'broom',
            price: 250,
            count: 1,
            category: 'Household Items',
          },
        ],
        id: 2,
        date: 1615122649963,
      },
    ],
  },
};

const axiosDataGetItemsResolved = {
  'Fruits & Vegatables': [{
    category: 'Fruits & Vegatables', count: 20, countInCart: 0, id: 1, name: 'apple', price: 120, src: 'https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80',
  }],
  'Household Items': [{
    category: 'Household Items', count: 3, countInCart: 0, id: 2, name: 'table cloth', price: 200, src: 'https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80',
  }],
};
const axiosDataGetOrdersResolved = [{
  date: 1615122360481,
  id: 1,
  items: {
    'Fruits & Vegatables': [{
      category: 'Fruits & Vegatables', count: 1, id: 1, name: 'apple', price: 120,
    }],
  },
}, {
  date: 1615122649963,
  id: 2,
  items: {
    'Household Items': [{
      category: 'Household Items', count: 1, id: 2, name: 'table cloth', price: 200,
    }, {
      category: 'Household Items', count: 1, id: 9, name: 'broom', price: 250,
    }],
  },
}];
export {
  axiosDataGetItems,
  axiosDataGetOrders,
  axiosDataGetItemsResolved,
  axiosDataGetOrdersResolved,
};
