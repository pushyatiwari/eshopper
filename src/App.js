/* eslint-disable no-console */
/* eslint-disable import/no-named-as-default */
/* eslint-disable react/react-in-jsx-scope */
import './App.css';
import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Home from './components/Home/Home';
import Navbar from './components/Navbar/Navbar';
import Cart from './components/Cart/Cart';
import Checkout from './components/Checkout/Checkout';
import AllOrders from './components/AllOrders/AllOrders';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      orders: [
        {
          orderId: 'o1',
          items: 12,
          orderDate: '12-12-2020',
          amount: 560.00,
          products: [
            {
              id: 'b1',
              name: 'Banana - Robusta',
              price: 40,
              count: 1,
            },
            {
              id: 'b2',
              name: 'melon',
              price: 40,
              count: 2,

            },
            {
              id: 'b3',
              name: 'apple',
              price: 40,
              count: 3,
            },
          ],
        },
        {
          orderId: 'o2',
          items: 12,
          orderDate: '12-12-2020',
          amount: 560.00,
          products: [
            {
              id: 'b1',
              name: 'Banana - Robusta',
              price: 40,
              count: 1,
            },
            {
              id: 'b2',
              name: 'melon',
              price: 40,
              count: 2,

            },
            {
              id: 'b3',
              name: 'apple',
              price: 40,
              count: 3,
            },
          ],
        }],
      products: [
        {
          id: 'b1',
          name: 'Banana - Robusta',
          price: 40,
          count: 0,
          src: 'https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png',
        },
        {
          id: 'b2',
          name: 'melon',
          price: 40,
          count: 0,
          src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRhOqYV2NWv0m80WzwGG2fCz6c3PzHmpz1Dg&usqp=CAU',

        },
        {
          id: 'b3',
          name: 'apple',
          price: 40,
          count: 0,
          src: 'https://d2lnr5mha7bycj.cloudfront.net/product-image/file/large_b0009f74-ef8e-48b4-b731-419ca93d8604.jpg',

        },
        {
          id: 'b4',
          name: 'item4',
          price: 40,
          count: 0,
          src: 'https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png',

        },
        {
          id: 'b5',
          name: 'Banana - Robusta',
          price: 40,
          count: 0,
          src: 'https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png',
        },
        {
          id: 'b6',
          name: 'melon',
          price: 40,
          count: 0,
          src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRhOqYV2NWv0m80WzwGG2fCz6c3PzHmpz1Dg&usqp=CAU',

        },
        {
          id: 'b7',
          name: 'apple',
          price: 40,
          count: 0,
          src: 'https://d2lnr5mha7bycj.cloudfront.net/product-image/file/large_b0009f74-ef8e-48b4-b731-419ca93d8604.jpg',

        },
        {
          id: 'b8',
          name: 'item4',
          price: 40,
          count: 0,
          src: 'https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png',

        },
      ],
      cartCount: 0,
    };
  }

  // onIncrement = (id) => {
  //   const { cartCount, products } = this.state;
  //   const newState = {
  //     ...this.state,
  //     cartCount: cartCount + 1,
  //     products: products.map((product) => {
  //       if (product.id === id) {
  //         return { ...product, count: product.count + 1 };
  //       }
  //       return product;
  //     }),
  //   };
  //   console.log(newState);
  //   this.setState(newState);
  // }

  onIncrement = (id) => {
    const { products } = this.state;
    this.setState((prevState) => {
      const newState = {
        ...prevState,
        products: products.map((product) => {
          if (product.id === id) {
            return { ...product, count: product.count + 1 };
          }
          return product;
        }),
        cartCount: prevState.cartCount + 1,

      };
      return newState;
    });
  }

  onDecrement = (id) => {
    const { cartCount, products } = this.state;

    if (cartCount <= 0) {
      return;
    }
    let flag = 0;
    products.forEach((product) => {
      if (product.id === id) {
        if (cartCount && product.count === 0) flag = 1;
      }
    });
    if (flag === 1) return;
    const newState = {
      ...this.state,
      cartCount: cartCount - 1,
      products: products.map((product) => {
        if (product.id === id) {
          if (product.count > 0) return { ...product, count: product.count - 1 };
          return { ...product };
        }
        return product;
      }),
    };
    console.log(newState);
    this.setState(newState);
  }

  render() {
    const { products, cartCount, orders } = this.state;
    return (
      <BrowserRouter>
        <Navbar cartCount={cartCount} />
        <Switch>
          <Route path="/" exact>
            <div className="container">
              <Home
                description={products}
                cartCount={cartCount}
                onIncrement={this.onIncrement}
                onDecrement={this.onDecrement}
              />
            </div>
          </Route>
          <Route path="/cart" render={() => <Cart description={products} cartCount={cartCount} />} />
          <Route path="/checkout">
            <Checkout />
          </Route>
          <Route path="/allorders">
            <AllOrders orders={orders} />
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }
}

export default App;
