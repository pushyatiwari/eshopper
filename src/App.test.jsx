// eslint-disable-next-line no-unused-vars
import {
  render, screen, fireEvent, waitFor, act,
} from '@testing-library/react';
import React from 'react';
import axios from 'axios';
import App from './App';
import { axiosDataGetOrders } from './mockData/mockData';

describe(App.name, () => {
  const axiosDataGetItems = {
    data: {
      data: [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 20,
          category: 'Fruits & Vegatables',
        },
      ],
    },

  };
  beforeEach(() => {
    jest.spyOn(axios, 'get')
      .mockResolvedValueOnce(
        axiosDataGetItems,
      )
      .mockResolvedValueOnce(axiosDataGetOrders)
      .mockResolvedValueOnce(null);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should match the snapshot', async () => {
    const { container } = await waitFor(() => render(<App />));
    expect(container).toMatchSnapshot();
  });

  test('should render a component Home', async () => {
    await waitFor(() => render(<App />));
    const clickEshopper = screen.getByText('E-shopper');
    fireEvent.click(clickEshopper);
    expect(document.location.href).toBe('http://localhost/');
  });
  test('should render component cart', async () => {
    await waitFor(() => render(<App />));
    const clickMyBasket = screen.getByTestId('cart-test-nav');
    fireEvent.click(clickMyBasket);
    expect(document.location.href).toBe('http://localhost/cart');
  });
  test('should increment cart count', async () => {
    await waitFor(() => render(<App />));
    const clickEshopper = screen.getByText('E-shopper');
    act(() => { fireEvent.click(clickEshopper); });
    const incrementButton = screen.getByText('+');
    fireEvent.click(incrementButton);
  });
});
