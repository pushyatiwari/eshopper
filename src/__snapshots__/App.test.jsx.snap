// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <header
    class="whiteTheme header"
  >
    <a
      class="e-shopper"
      href="/"
    >
      <p>
        E-shopper
      </p>
    </a>
    <div
      class="nav-order-details"
    >
      <a
        href="/allorders"
      >
        <button
          class="nav-buttons"
        >
          All Orders
        </button>
      </a>
      <a
        data-testid="cart-test-nav"
        href="/cart"
      >
        <button
          class="whiteTheme nav-buttons"
        >
          My Basket :
          0
        </button>
      </a>
    </div>
  </header>
  <button
    type="button"
  >
    Change Theme
  </button>
  <div
    class="container"
  >
    <div
      class="whiteTheme container"
      data-testid="home-route"
    >
      <div
        class="rendercards"
      >
        <div
          class="productContainer"
          data-testid="product"
        >
          <div
            class="product"
          >
            <p>
              Fruits & Vegatables
            </p>
            <img
              alt="product"
              src="https://images.unsplash.com/photo-1571771894821-ce9b6c11b08e?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1000&q=80"
            />
            <p
              class="productDescription"
            >
              apple
            </p>
            <p
              class="productDescription"
            >
              1 kg
            </p>
          </div>
          <div
            class="description"
          >
            <p>
              MRP:
              120
            </p>
            <div
              class="basketItems"
            >
              <button
                type="button"
              >
                -
              </button>
              0
              <button
                type="button"
              >
                +
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
